web/src/components/vis/mixins/plotData.js

```diff
@@ -0,0 +1,54 @@
+import { mapState, mapActions, mapMutations } from 'vuex';
+import { LOAD_PLOT } from '../../../store/actions.type';
+import { SET_PLOT } from '../../../store/mutations.type';
+
+/**
+ * Expects `id` to exist on the wrapping component
+ * @param {String} plotName name of the plot to subscribe to
+ */
+export default function (plotName) {
+  return {
+    data() {
+      return {
+        active: true,
+      };
+    },
+    activated() { this.active = true; },
+    deactivated() { this.active = false; },
+    mounted() { this.reloadPlot(); },
+    computed: {
+      ...mapState(['loading']),
+      dataset() { return this.$store.getters.dataset(this.id); },
+      plot() { return this.$store.getters.plot(this.id, plotName); },
+    },
+    watch: {
+      // eslint-disable-next-line func-names
+      'plot.valid': function () { this.reloadPlot(); },
+      id() { this.reloadPlot(); },
+      active(val) {
+        if (val) {
+          this.reloadPlot();
```

Comment:
Is this meant to solve the problem of things not updating after making changes earlier in the pipeline? Does this solve the ["sticky vis" problem](https://github.com/girder/viime/issues/215)?

Reply:
Yes, this should(?) solve that problem.